### PR TITLE
🌱 Drop dependency on x/exp/slices

### PR DESCRIPTION
**Pull request by dtantsur**
Go 1.21 already has what we need in the standard library.

Signed-off-by: Dmitry Tantsur <dtantsur@protonmail.com>

### CODE REVIEW

**Comment by lentzi90** [comment]
I cannot reproduce the test failure locally. My best guess is something was not properly cleaned up between runs.
/test metal3-bmo-e2e-test-pull

**Review by lentzi90 — COMMENTED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -60,6 +59,7 @@ require (
 	github.com/spf13/cobra v1.7.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
+	golang.org/x/exp v0.0.0-20230522175609-2e198f4a06a1 // indirect
```
Comment (kashifest) [inline go.mod:62]: Which module is still dependant on this?

Comment (dtantsur) [reply go.mod:62]: I'm not sure how to figure that out. Nothing in our code, I believe.

Comment (kashifest) [reply go.mod:62]: `go mod graph | grep "golang.org/x/exp"` should give us the module which is using this.

Comment (kashifest) [reply go.mod:62]: ok its coming from 

`k8s.io/apiextensions-apiserver@v0.28.3` and 
`sigs.k8s.io/controller-runtime@v0.16.3`, so we cant really get rid of this entry.

/approve

Comment (dtantsur) [reply go.mod:62]: TIL `go mod graph`!

**Pull request merged**